src/unitxt/loaders.py

```diff
@@ -62,13 +70,24 @@ class LoadFromIBMCloud(Loader):
     data_files: Sequence[str]
 
     def _download_from_cos(self, cos, bucket_name, item_name, local_file):
-        print(f"Downloading {item_name} from {bucket_name} COS to {local_file}")
+        print(f"Downloading {item_name} from {bucket_name} COS")
         try:
             response = cos.Object(bucket_name, item_name).get()
             size = response["ContentLength"]
+            body = response["Body"]
         except Exception as e:
             raise Exception(f"Unabled to access {item_name} in {bucket_name} in COS", e)
 
+        if self.loader_limit != None:
```

Comment:
you are still downloading everything, no? What am I missing?

Reply:
Loader limit is set to each of the loaders by the calling recipe.